code/lib/core-server/src/utils/new-story-templates/typescript.test.ts

```diff
@@ -0,0 +1,56 @@
+import { describe, expect, it } from 'vitest';
+import { getTypeScriptTemplateForNewStoryFile } from './typescript';
+
+describe('typescript', () => {
+  it('should return a TypeScript template with a default import', () => {
+    const result = getTypeScriptTemplateForNewStoryFile({
+      basename: 'foo',
+      componentExportName: 'default',
+      default: true,
+      frameworkPackageName: '@storybook/nextjs',
+      exportedStoryName: 'Default',
+    });
+
+    expect(result).toMatchInlineSnapshot(`
+      "import type { Meta, StoryObj } from '@storybook/nextjs';
+
+      import Component from './foo';
+
+      const meta = {
+        component: Component
+      } satisfies Meta<typeof Component>
+
+      export default meta;
+
+      type Story = StoryObj<typeof meta>;
+
+      export const Default: Story = {}"
+    `);
+  });
+
+  it('should return a TypeScript template with a named import', () => {
+    const result = getTypeScriptTemplateForNewStoryFile({
+      basename: 'foo',
+      componentExportName: 'Example',
+      default: false,
+      frameworkPackageName: '@storybook/nextjs',
+      exportedStoryName: 'Default',
+    });
+
+    expect(result).toMatchInlineSnapshot(`
+      "import type { Meta, StoryObj } from '@storybook/nextjs';
+
+      import { Example } from './foo';
+
+      const meta = {
+        component: Example
+      } satisfies Meta<typeof Example>
+
+      export default meta;
+
+      type Story = StoryObj<typeof meta>;
+
+      export const Default: Story = {}"
```

Comment:
Missing a newline at end-of-file.

Reply:
This also falls into the (opinionated) formatting preferences of the user. As elaborated earlier, we don't have a proper solution in place to format the file by using the user's formatter and formatting rules.